src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/MarkupCompiler/MarkupCompiler.cs

```diff
@@ -2596,7 +2603,20 @@ private void GenerateInitializeComponent(bool isApp)
 
             string uriPart = string.Empty;
 
-            string version = String.IsNullOrEmpty(AssemblyVersion) ? String.Empty : COMPONENT_DELIMITER + VER + AssemblyVersion;
+            bool hasWildcard = false;
+
+            // Attempt to parse out the AssemblyVersion if it exists.  This validates that we can either use an empty version string (wildcards exist)
+            // or we can utilize the passed in string (valid parse).
+            if (!string.IsNullOrEmpty(AssemblyVersion)
+                && !VersionHelper.TryParseAssemblyVersion(AssemblyVersion, allowWildcard: true, out Version _, out hasWildcard))
+            {
+                throw new AssemblyVersionParseException(SR.Get(SRID.InvalidAssemblyVersion, AssemblyVersion));
+            }
+
+            // If a developer explicitly sets the AssemblyVersion build property to a wildcard version string, we would use that as part of the URI here.
+            // This results in an error in Version.Parse during InitializeComponent's call tree.  Instead, do as we would have when the developer sets a
+            // wildcard version string via AssemblyVersionAttribute and use an empty string.
+            string version = (hasWildcard || String.IsNullOrEmpty(AssemblyVersion)) ? String.Empty : COMPONENT_DELIMITER + VER + AssemblyVersion;
```

Comment:
outer parens may be unnecessary 

nit: for better readability, consider folding ternary operator into multiple lines, like this: 

```c#
            string version = hasWildcard || String.IsNullOrEmpty(AssemblyVersion) 
                ? String.Empty 
                : COMPONENT_DELIMITER + VER + AssemblyVersion;
```